lib/core/prometheusMetrics.hpp

```diff
@@ -16,6 +16,116 @@ using std::string;
 
 namespace kotekan {
 
+/**
+ * @class metric
+ * @brief An internal base class for storing metric values
+ */
+class metric {
+public:
+    /// @brief Returns the stored value as a string.
+    virtual string to_string() = 0;
+    virtual ~metric() = default;
+
+protected:
+    /// The actual value to be returned
+    double value = 0;
+
+    /// Internal function to get the time in
+    static uint64_t get_time_in_milliseconds();
+};
+
+class gauge : public metric {
+public:
+    /// @brief Sets the current metric value
+    void set(double value);
+    string to_string() override;
+private:
+    /// Time stamp in milliseconds.
+    uint64_t last_update_time_stamp;
+};
+
+/**
+ * @class metric
+ * @brief An internal base class for storing metric values
+ */
+class Metric {
+public:
+    Metric(const std::vector<string>&);
+    virtual ~Metric() = default;
+
+    /// @brief Returns the stored value as a string.
+    virtual string to_string() = 0;
+    virtual std::ostringstream& to_string(std::ostringstream& out) = 0;
+    const std::vector<string> label_values;
+};
+
+class Counter : public Metric {
+public:
+    Counter(const std::vector<string>&);
+    void inc();
```

Comment:
Tempted to operator overload `++` here.

Reply:
I'm leaving it alone for now, since `.inc()` is used in other Prometheus clients. Let's discuss this and other ways to improve the API from developers' point after this cycle.